src/components/progress-bar/progress-bar.ts

```diff
@@ -0,0 +1,132 @@
+import {
+    Component,
+    ChangeDetectionStrategy,
+    HostBinding,
+    Input,
+} from 'angular2/core';
+import {isPresent, CONST} from 'angular2/src/facade/lang';
+import {OneOf} from '../../core/annotations/one-of';
+
+
+// TODO(josephperrott): Benchpress tests.
+// TODO(josephperrott): Add ARIA attributes for progressbar "for".
+
+
+/** Display modes of Progress Bar*/
+@CONST()
+class ProgressMode {
+  @CONST() static DETERMINATE = 'determinate';
+  @CONST() static INDETERMINATE = 'indeterminate';
+  @CONST() static BUFFER = 'buffer';
+  @CONST() static QUERY = 'query';
+}
+
+
+/**
+ * <md-progress-bar> component.
+ */
+@Component({
+  selector: 'md-progress-bar',
+  host: {
+    'role': 'progressbar',
+    'aria-valuemin': '0',
+    'aria-valuemax': '100',
+  },
+  templateUrl: './components/progress-bar/progress-bar.html',
+  styleUrls: ['./components/progress-bar/progress-bar.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class MdProgressBar {
+  /**
+   * Value of the progressbar.
+   *
+   * Defaults to zero. Mirrored to aria-valuenow.
+   */
+  @Input('value')
```

Comment:
Not sure I understand what you are saying here.

I dont think it makes sense to put `@Input` on a getter.

It currently is:

```
@Input('value') _value: number = 0;
@HostBinding('attr.aria-valuenow') get value_() { return this._value; }
```